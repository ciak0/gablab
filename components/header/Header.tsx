import type { FunctionComponent } from 'react';
import { differenceInYears } from 'date-fns';
import gab from '../../public/gab.jpg';
import { DATE_OF_BIRTH, LINKED_IN_URL } from '../../constants';
import LocalImage from '../local-image/LocalImage';

const Header: FunctionComponent = () => (
  <header className="p-4">
    <p className="text-center">
      This is Gab, I am a
      {' '}
      <strong>
        {differenceInYears(new Date(), DATE_OF_BIRTH)}
      </strong>
      {' '}
      years old human.
      <br />
      I&apos;ve been writing software since
      {' '}
      <strong>2001</strong>
      , professionnally since
      {' '}
      <strong>2006</strong>
      .
      <br />
      I am a
      {' '}
      <a
        href="https://www.4dayweek.com/"
        target="_blank"
        rel="noreferrer"
        className="underline"
      >
        4 day week
      </a>
      {' '}
      ambassador since 2013.
    </p>

    <div className="py-8 text-center next-img-overflow-visible">
      <LocalImage
        className="rounded-full border border-gray-300 shadow-neutral-500 shadow-md"
        src={gab}
        alt="Gab"
        width={256}
        height={256}
      />
    </div>

    <p className="text-center">
      Here you can find my CV, feel free to contact me on
      {' '}
      <a
        href={LINKED_IN_URL}
        target="_blank"
        className=" text-blue-600 underline"
        rel="noreferrer"
      >
        LinkedIn
      </a>
      .
    </p>
  </header>
);

export default Header;
