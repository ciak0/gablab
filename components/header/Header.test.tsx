import { render } from '@testing-library/react';
import Header from './Header';

describe('components/Header', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Header />,
    );

    expect(container).toMatchSnapshot();
  });
});
