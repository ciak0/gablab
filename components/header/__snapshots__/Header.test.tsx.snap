// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Header should render correctly 1`] = `
<header
  className="p-4"
>
  <p
    className="text-center"
  >
    This is Gab, I am a
     
    <strong>
      39
    </strong>
     
    years old human.
    <br />
    I've been writing software since
     
    <strong>
      2001
    </strong>
    , professionally since
     
    <strong>
      2006
    </strong>
    .
    <br />
    I am a
     
    <a
      className="underline"
      href="https://www.4dayweek.com/"
      rel="noreferrer"
      target="_blank"
    >
      4 day week
    </a>
     
    ambassador since 2013.
  </p>
  <div
    className="py-8 text-center next-img-overflow-visible"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjU2IiBoZWlnaHQ9IjI1NiIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="Gab"
        className="rounded-full border border-gray-300 shadow-neutral-500 shadow-md"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="Gab"
          className="rounded-full border border-gray-300 shadow-neutral-500 shadow-md"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="test-file-stub"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
  <p
    className="text-center"
  >
    Here you can find my CV, feel free to contact me on
     
    <a
      className=" text-blue-600 underline"
      href="https://www.linkedin.com/in/gabriele-ferrari-a5095b52/"
      rel="noreferrer"
      target="_blank"
    >
      LinkedIn
    </a>
    .
  </p>
</header>
`;
